### PR TITLE
[5.3] Cleanup authorize middleware

### PR DESCRIPTION
**Pull request by JosephSilber**
Smaller functions that are more self explanatory, and don't need a big inline comment to explain what's going on.

### CODE REVIEW

File: src/Illuminate/Auth/Middleware/Authorize.php [hunk 1]
```diff
@@ -65,17 +65,29 @@ public function handle($request, Closure $next, $ability, $model = null)
...
+     * @return string|\Illuminate\Database\Eloquent\Model
+     */
+    protected function getModel($request, $model)
+    {
+        return $this->isClassName($model) ? $model : $request->route($model);
+    }
 
-        return $request->route($model);
+    /**
+     * Checks if the given string looks like a fully qualified class name.
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Middleware/Authorize.php]: Note that fully qualified class names are actually defined **without** a leading slash. The leading slash is just like an escape character.

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by taylorotwell** [comment]
Any reason to not just use `class_exists` instead of that `\` check? Current implementation would fail on global namespaced classes. Which I sometimes use when playing with stuff in the routes.php file.

**Comment by JosephSilber** [comment]
I guess we could do that. I just thought that it's a pity to load another file if we don't need it.

Alternatively we could check if the first letter is capitalized. That will break for people that capitalize their ability names (never seen that), but so might `class_exists` if they have matching non-namespaced classes.

Overall I believe the current implementation to be the safest one, but feel free to change it if you feel the need.

File: src/Illuminate/Auth/Middleware/Authorize.php [hunk 2]
```diff
@@ -65,17 +65,29 @@ public function handle($request, Closure $next, $ability, $model = null)
      */
```
Comment (lilblue31) [inline src/Illuminate/Auth/Middleware/Authorize.php]: d0:5b:a8:31:d0:8e1000000000000000000000000

**Pull request merged**